### PR TITLE
Devoncarew fail on errors

### LINKED ISSUE #489: dartdoc should fail to generate docs if any analysis errors
It should probably (possibly?) still generate docs, but should warn the user at the end that there were analysis errors encountered:

```
"4 analysis errors and 123 warnings encountered!"
```

### PR DESCRIPTION
**Pull request by devoncarew**
fix #489

- calculate the analysis errors for the package
- fail (`exit(1)`) if there are any error level issues
- print out the list of errors, with file path and line numbers

### CODE REVIEW

File: lib/dartdoc.dart [hunk 1]
```diff
@@ -120,15 +124,45 @@ class DartDoc {
...
+    for (Source source in sources) {
+      context.computeErrors(source);
+      errorInfos.add(context.getErrors(source));
+    }
+
+    List<_Error> errors = errorInfos.expand((AnalysisErrorInfo info) {
+      return info.errors.map(
+          (error) => new _Error(error, info.lineInfo, packageMeta.dir.path));
+    }).where((_Error error) => error.isError).toList();
+    errors.sort();
```
Comment (sethladd) [inline lib/dartdoc.dart]: you could do:

`..sort()`

**Comment by sethladd** [comment]
Can we add a test for this?

File: lib/dartdoc.dart (hunk 1 above)
Comment (devoncarew) [inline lib/dartdoc.dart]: done

**Comment by devoncarew** [comment]
> Can we add a test for this?

Yup; I'll need to do some refactoring so we can test this, otherwise the call to `exit(1)` will be a bit of an issue.

**Comment by sethladd** [comment]
Thanks for adding the test!

**Pull request merged**